bin/copyQCDataToTmp.sh

```diff
@@ -0,0 +1,389 @@
+CFG_DIR="${INSTALLATION_DIR}/etc"
+HOSTNAME_SHORT="$(hostname -s)"
+ROLE_USER="$(whoami)"
+REAL_USER="$(logname 2>/dev/null || echo 'no login name')"
+
+#
+##
+### Functions.
+##
+#
+
+if [[ -f "${LIB_DIR}/sharedFunctions.bash" && -r "${LIB_DIR}/sharedFunctions.bash" ]]
+then
+	# shellcheck source=lib/sharedFunctions.bash
+	source "${LIB_DIR}/sharedFunctions.bash"
+else
+	printf '%s\n' "FATAL: cannot find or cannot access sharedFunctions.bash"
+	trap - EXIT
+	exit 1
+fi
+
+function copyQCRawdataToTmp() {
+
+	local _rawdata="${1}"
+	log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Working on ${_rawdata}"
+
+	if [[ -e "${PRM_ROOT_DIR}/rawdata/ngs/${_rawdata}/Info/SequenceRun.csv" ]]
+	then
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Sequencerun ${_rawdata} is not yet copied to tmp, start rsyncing.."
+		touch "${JOB_CONTROLE_FILE_BASE}.started"
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${_rawdata} found on ${PRM_ROOT_DIR}, start rsyncing.."
+		rsync -av --rsync-path="sudo -u ${group}-ateambot rsync" "${PRM_ROOT_DIR}/rawdata/ngs/${_rawdata}/Info/SequenceRun"* "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/trendanalysis/rawdata/${_rawdata}/" \
+		|| {
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Failed to rsync ${line}"
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    from /groups/${group}/${PRM_ROOT_DIR}/"
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    to ${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/"
+		mv "${JOB_CONTROLE_FILE_BASE}."{started,failed}
+		return
+		}
+
+		mv "${JOB_CONTROLE_FILE_BASE}."{started,finished}
+	else
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "For sequencerun ${_rawdata} there is no QC data, nothing to rsync.."
+	fi
+
+}
+
+function copyQCProjectdataToTmp() {
+
+	local _project="${1}"
+	log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Working on ${_project}"
+
+	if [[ -e "${PRM_ROOT_DIR}/projects/${_project}/run01/results/multiqc_data/${_project}.run_date_info.csv" ]]
+	then
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Project ${_project} is not yet copied to tmp, start rsyncing.."
+		touch "${JOB_CONTROLE_FILE_BASE}.started"
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${_project} found on ${PRM_ROOT_DIR}, start rsyncing.."
+		rsync -av --rsync-path="sudo -u ${group}-ateambot rsync" "${PRM_ROOT_DIR}/projects/${_project}/run01/results/multiqc_data/"* "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/trendanalysis/projects/${_project}/" \
+		|| {
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Failed to rsync ${line}"
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    from /groups/${group}/${PRM_ROOT_DIR}/"
+		log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    to ${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/"
+		mv "${JOB_CONTROLE_FILE_BASE}."{started,failed}
+		return
+		}
+
+		mv "${JOB_CONTROLE_FILE_BASE}."{started,finished}
```

Comment:
I think you need something like this instead
`local _controlFileBaseForFunction="${_controlFileBase}.${FUNCNAME[0]}"` of `${JOB_CONTROLE_FILE_BASE}` (examples of this are in other NGS_Automated scripts)
Because now you are already touching .finished for the script and the Darwin part needs to start after this function

Reply:
controlFileBase="${PRM_ROOT_DIR}/trendanalysis/logs/${rawdata}/"
export JOB_CONTROLE_FILE_BASE="${controlFileBase}/${rawdata}.${SCRIPT_NAME}"

de JOB_CONTROLE_FILE_BASE in elke functie heeft de ruwdata/projectdata/darwindata ook in de naam. de JOB_CONTROLE_FILE_BASE is 200210_NB501093_0124_AHWFVCAFXY.copyQCDataToTmp.finished voor de ruwedata. Voor de darwin data komt er een nieuwe JOB_CONTROLE_FILE_BASE. of bedoel je dit niet?